### PR TITLE
✨ clusterctl: add command groups

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Improves the clusterctl command overview by grouping the commands. I opted to create two groups: "Management commands" and "Other commands". **I welcome any feedback regarding the names.**

This feature would also allow the flattening of the command structure in the future. For example the alpha commands could be placed under a "Alpha commands" group and displayed immediately under the help command. 

New clusterctl output:
```
Cluster Management Commands:
  delete      Delete one or more providers from the management cluster
  generate    Generate yaml using clusterctl yaml processor
  get         Get info from a management or workload cluster
  init        Initialize a management cluster
  move        Move Cluster API objects and all dependencies between management clusters
  upgrade     Upgrade core and provider components in a management cluster

Troubleshooting and Debugging Commands:
  describe    Describe workload clusters

Other Commands:
  alpha       Commands for features in alpha
  completion  Output shell completion code for the specified shell (bash or zsh)
  config      Display clusterctl configuration
  help        Help about any command
  version     Print clusterctl version
```

Old clusterctl output:
```
Available Commands:
  alpha       Commands for features in alpha.
  backup      Backup Cluster API objects and all dependencies from a management cluster.
  completion  Output shell completion code for the specified shell (bash or zsh)
  config      Display clusterctl configuration.
  delete      Delete one or more providers from the management cluster.
  describe    Describe workload clusters.
  generate    Generate yaml using clusterctl yaml processor.
  get         Get info from a management or a workload cluster
  help        Help about any command
  init        Initialize a management cluster.
  move        Move Cluster API objects and all dependencies between management clusters.
  restore     Restore Cluster API objects from file by glob. Object files are searched in config directory
  upgrade     Upgrade core and provider components in a management cluster.
  version     Print clusterctl version.
```

### CODE REVIEW

File: cmd/clusterctl/cmd/root.go [hunk 1]
```diff
@@ -118,6 +123,19 @@ func init() {
 	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "",
 		"Path to clusterctl configuration (default is `$HOME/.cluster-api/clusterctl.yaml`) or to a remote location (i.e. https://example.com/clusterctl.yaml)")
 
+	RootCmd.AddGroup(
+		&cobra.Group{
+			ID:    groupManagement,
+			Title: "Management commands:",
```
Comment (fabriziopandini) [inline cmd/clusterctl/cmd/root.go:129]:
```suggestion
			Title: "Cluster Management Commands:",
```
For consistency with kubectl

File: cmd/clusterctl/cmd/alpha.go [hunk 2]
```diff
@@ -21,9 +21,10 @@ import (
 )
 
 var alphaCmd = &cobra.Command{
-	Use:   "alpha",
-	Short: "Commands for features in alpha",
-	Long:  `These commands correspond to alpha features in clusterctl.`,
+	Use:     "alpha",
+	GroupID: groupManagement,
```
Comment (fabriziopandini) [inline cmd/clusterctl/cmd/alpha.go:25]:
```suggestion
	GroupID: groupOther,
```
for consistency with kubectl

File: cmd/clusterctl/cmd/root.go [hunk 3]
```diff
@@ -118,6 +123,19 @@ func init() {
 	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "",
 		"Path to clusterctl configuration (default is `$HOME/.cluster-api/clusterctl.yaml`) or to a remote location (i.e. https://example.com/clusterctl.yaml)")
 
+	RootCmd.AddGroup(
```
Comment (fabriziopandini) [inline cmd/clusterctl/cmd/root.go:126]: Q: what about creating a third group "Troubleshooting and Debugging Commands:" with describe (in future we will move also topology plan as soon as cluster class graduates to beta)

File: cmd/clusterctl/cmd/config.go [hunk 4]
```diff
@@ -21,9 +21,10 @@ import (
 )
 
 var configCmd = &cobra.Command{
-	Use:   "config",
-	Short: "Display clusterctl configuration",
-	Long:  `Display clusterctl configuration.`,
+	Use:     "config",
+	GroupID: groupManagement,
```
Comment (fabriziopandini) [inline cmd/clusterctl/cmd/config.go:25]:
```suggestion
	GroupID: groupOther,
```
for consistency with kubectl

**Review by fabriziopandini — COMMENTED** [review]
@oscr that's a nice improvement, thanks!
added a few of suggestions, let's hear also other opinions 
cc @ykakarap and @Jont828

File: cmd/clusterctl/cmd/root.go (hunk 3 above)
Comment (oscr) [reply cmd/clusterctl/cmd/root.go:126]: I think it sounds like a good idea. What command would you like to see in this group?

Comment (fabriziopandini) [reply cmd/clusterctl/cmd/root.go:126]: only describe for now

File: cmd/clusterctl/cmd/root.go [hunk 5]
```diff
@@ -118,6 +123,19 @@ func init() {
...
+		&cobra.Group{
+			ID:    groupManagement,
+			Title: "Cluster Management Commands:",
+		},
+		&cobra.Group{
+			ID:    groupOther,
+			Title: "Other commands:",
```
Comment (fabriziopandini) [inline cmd/clusterctl/cmd/root.go:133]:
```suggestion
			Title: "Other Commands:",
```
Also the other group is written in with all the words in capital letters

File: cmd/clusterctl/cmd/root.go (hunk 3 above)
Comment (oscr) [reply cmd/clusterctl/cmd/root.go:127]: Fixed

**Comment by fabriziopandini** [comment]
/lgtm

@ykakarap @Jont828 PTAL

**Review by ykakarap — APPROVED** [review]
/lgtm

This is great. Thank you!

Right now the `alpha` commands are a bucket of everything. Example `alpha rollout` would best suite  "Cluster Management Commands" group and `alpha topology plan` would best suite "Troubleshooting and Debugging Commands" group. But since in this itearation we are only doing top level grouping the current grouping we have makes sense.

**Comment by fabriziopandini** [comment]
Great Stuff!
/lgtm
/approve

**Comment by oscr** [comment]
@ykakarap @fabriziopandini Would you like me to also make prs for 1.3 and 1.2?

**Comment by sbueringer** [comment]
In general we don't backport features https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#backporting-a-patch

**Comment by sbueringer** [comment]
Nice feature, thx!!

**Pull request merged**